### ISSUE TITLE
Error Handling

### ISSUE THREAD

**Issue by jafri** [EOSIO/eosjs#708]
I think eosjs needs a substantial update that provides clear error messages. Currently, it just passes nodeos errors on, but these could be substantially improved, especially in the case of CPU where the error is ambiguous and often incomprehensible to users.

**Comment by jlamarr22** [comment]
I agree with you that there are some instances where it's difficult to figure out how to fix errors due to ambiguous messages. 
> Currently, it just passes nodeos errors on, but these could be substantially improved

This sounds like something that could be improved on nodeos. If we start translating error messages on the EOSJS side, then do we translate all of them? How do we draw the line between error messages we want to translate and ones we don't need to? It doesn't seem like this solution would solve the root cause of the issue.

If you have a set of specific scenarios that are causing strange error messages and reasons why they don't make sense to you -- I can reach out to others and see if they can be improved on the nodeos side.

**Comment by jafri** [comment]
@jlamarr22 the largest problems are around resource management. Error messages with clear actionable steps and helper information like "Stake 5 more EOS for TX to pass", are much needed.
